### PR TITLE
CAdES: Fix SignerInfo attribute construction order.

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
Small fixup on the recent CAdES contribution.
Can be trigger if someone is using CMS_REUSE_DIGEST flag value.
Not a big issue, just a lack of logic ;)

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by FdaSilvaYY** [comment]
Ping @openssl: small fix quick to review.
The CAdES code addition was  one code block too far, my mistake.
I improve a bit asn1 fuzz test at the same time.

File: crypto/cms/cms_sd.c [hunk 1]
```diff
@@ -353,6 +346,13 @@ CMS_SignerInfo *CMS_add1_signer(CMS_ContentInfo *cms,
             if (!add_sc)
                 goto err;
         }
+        if (flags & CMS_REUSE_DIGEST) {
```
Comment (FdaSilvaYY) [inline crypto/cms/cms_sd.c]: This code-block was just move down.

**Comment by slontis** [comment]
The travis failures are relevant.

File: fuzz/asn1.c [hunk 2]
```diff
@@ -292,8 +293,7 @@ int FuzzerTestOneInput(const uint8_t *buf, size_t len)
 {
     int n;
 
-
-    for (n = 0; item_type[n] != NULL; ++n) {
+    for (n = 0; n < OSSL_NELEM(item_type); ++n) {
```
Comment (slontis) [inline fuzz/asn1.c]: Not sure why this needed changing - the NULL check was ok.

Comment (slontis) [reply fuzz/asn1.c]: I would probably revert these NULL check changes.

**Comment by FdaSilvaYY** [comment]
@slontis : the CAdES verify feature changes and some additional tests are here : #8098 
I will add the suggested tests in it.

File: fuzz/asn1.c (hunk 2 above)
Comment (FdaSilvaYY) [reply fuzz/asn1.c]: I prefer something more straightforward as a size deduced by compiler than expecting a NULL ending item.

**Comment by slontis** [comment]
@FdaSilvaYY is this code in 1_1 also?

**Comment by slontis** [comment]
The travis failure is related to the failing extended test (which is a known failure currently)

**Comment by FdaSilvaYY** [comment]
> 
> 
> @FdaSilvaYY is this code in 1_1 also?

No, #7893 was just merged into master

**Comment by slontis** [comment]
Merged. Thanks @FdaSilvaYY. You probably will want to rebase your other PR now that this one has gone in.

**Pull request closed without merging**